artemis-server/src/main/java/org/apache/activemq/artemis/core/persistence/impl/journal/OperationContextImpl.java

```diff
@@ -415,4 +449,27 @@ public String toString() {
          executorsPendingField +
          "]";
    }
+
+   @Override
+   public void clear() {
+      stored = 0;
+      storeLineUpField = 0;
+      minimalReplicated = 0;
+      replicated = 0;
+      replicationLineUpField = 0;
+      paged = 0;
+      minimalPage = 0;
+      pageLineUpField = 0;
+      errorCode = -1;
+      errorMessage = null;
+      executorsPendingField = 0;
+
+      if (tasks != null) {
+         tasks.clear();
+      }
+
+      if (storeOnlyTasks != null) {
+         storeOnlyTasks.clear();
+      }
```

Comment:
These lists are not thread-safe, and this could potentially be a concurrent operation (people can tend to use 'force' toggles even if they dont actually need to). Was this method meant to be synchronized, or did you not want to synchronize it (although setting the other fields also wont really be safe without doing so..) ? If the latter do they need changed to a concurrent structure, or is it ok they may get corrupted in this scenario?

Reply:
@gemmellr yes this could potentially be a concurrent operation and it could cause inconsistent state. I could change it to be thread-safe but I'm not sure this is necessary. I mean this method should be the last chance to close a stuck session and it is supposed to be manually called by a user. If the call is successful the context/session will be closed otherwise users could retry.

Reply:
So I guess the question is really just, what are the potential consequences of corrupting the list and is it important?

Closing 'stuck' sessions is what the method is being added for, but its not necessarily when it will actually be called. I expect some folks will use the 'force' just because it exists at all, to avoid having to think whether they actually need it or not. E.g the same way folks use rm -rf a lot of the time.

Reply:
The potential consequences of corrupting should only be related to the closing session itself and its context but I see your point I'll change it to be thread-safe.